cibyl/plugins/openstack/printers/serialized.py

```diff
@@ -0,0 +1,141 @@
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+"""
+from abc import ABC
+
+from overrides import overrides
+
+from cibyl.outputs.cli.printer import JSONPrinter, SerializedPrinter
+from cibyl.plugins.openstack import Deployment
+from cibyl.plugins.openstack.container import Container
+from cibyl.plugins.openstack.node import Node
+from cibyl.plugins.openstack.package import Package
+from cibyl.plugins.openstack.printers import OSPrinter
+from cibyl.plugins.openstack.service import Service
+
+
+class OSSerializedPrinter(OSPrinter, SerializedPrinter, ABC):
+    """Provides a machine-readable representation of the plugin's models for
+    easy readability from a machine.
+    """
+
+    @overrides
+    def print_deployment(self, deployment: Deployment) -> str:
+        network = deployment.network.value
+        storage = deployment.storage.value
+        ironic = deployment.ironic.value
+
+        result = {
+            'release': deployment.release.value,
+            'infra_type': deployment.infra_type.value,
+            'topology': deployment.topology.value,
+            'network': {
+                'ip_version': network.ip_version.value,
+                'network_backend': network.network_backend.value,
+                'ml2_driver': network.ml2_driver.value,
+                'security_group': network.security_group.value,
+                "dvr": network.dvr.value,
+                'tls_everywhere': network.tls_everywhere.value
+            },
+            'storage': {
+                'cinder_backend': storage.cinder_backend.value
+            },
+            'ironic': {
+                'ironic_inspector': ironic.ironic_inspector.value,
+                'cleaning_network': ironic.cleaning_network.value
+            },
+            'overcloud_templates': list(deployment.overcloud_templates.value),
+            'test_collection': self.provider.load(
+                self.print_test_collection(
+                    deployment.test_collection.value
+                )
+            ),
+            'nodes': [
+                self.provider.load(self.print_node(node))
+                for node in deployment.nodes.values()
+            ],
+            'services': [
+                self.provider.load(self.print_service(service))
+                for service in deployment.services.values()
+            ],
+            'stages': [
+                {
+                    'name': stage.name.value,
+                    'status': stage.status.value,
+                    'duration': stage.duration.value
+                }
+                for stage in deployment.stages
+            ]
+        }
+
+        return self.provider.dump(result)
+
+    @overrides
+    def print_test_collection(self, collection) -> str:
+        result = {
+            'tests': list(collection.tests.value),
```

Comment:
This could have the same issue as the overcloud_templates line

Reply:
Fixed.